volpe/annotate.py

```diff
@@ -111,7 +112,18 @@ def escaped_character(_: TypeTree):
 
     def string(self, tree: TypeTree):
         tree.data = "list"
-        text = eval(tree.children[0])
+        # Evaluate string using Python
+        try:
+            text = eval(tree.children[0])
+        except SyntaxError as err:
+            raise VolpeError(err.msg, tree)
+        # Check that all characters are valid ascii
+        if version_info >= (3, 7, 0):
```

Comment:
if the old method works on 3.7, then it would probably be cleaner to only use that

Reply:
I have not tested it myself, but from stackoverflow answers the 3.7 version is significantly faster.
I agree that it would be cleaner to just use one instead of doing things differently for different versions, but we should also think about optimisation when it's easy like this and still readable.

Maybe we should introduce `version_dependent.py` and have all the python version differences stored there, so then you would just `import isascii from version_dependent`.

Thoughts?

Reply:
hmm, that does sound like a nice solution

Reply:
Ok, I added it